### PR TITLE
Device runtime configuration (initiallly for MIDI devices)

### PR DESCRIPTION
**Pull request by Cacodemon345**
Summary
=======
This PR introduces device runtime configuration support. Initially done for RtMidi and FluidSynth MIDI backends.

Checklist
=========
* [ ] I have discussed this with core contributors already
* [ ] This pull request requires changes to the ROM set
  * [ ] I have opened a roms pull request - https://github.com/86Box/roms/pull/changeme/

References
==========
None.

### CODE REVIEW

File: src/qt/qt_deviceconfig.cpp [hunk 1]
```diff
@@ -365,6 +377,14 @@ DeviceConfig::ConfigureDevice(const _device_ *device, int instance, Settings *se
             }
             config++;
         }
+
+        /* The :: prefix is intentional. */
```
Comment (dhrdlicka) [inline src/qt/qt_deviceconfig.cpp:381]: Maybe explain _why_ it's there rather than just say that it's intended?

File: src/qt/qt_mainwindow.ui [hunk 2]
```diff
@@ -87,7 +87,14 @@
     <property name="title">
      <string>&amp;Tools</string>
     </property>
+    <widget class="QMenu" name="menuDevice_settings">
+     <property name="title">
+      <string>Device settings...</string>
```
Comment (dhrdlicka) [inline src/qt/qt_mainwindow.ui:92]:
```suggestion
      <string>Device settings</string>
```
There should be no ellipsis at the end, since it's a submenu.

**Review by dhrdlicka — DISMISSED** [review]

**Comment by dhrdlicka** [comment]
I think ideally this should just be implemented in the Settings dialog, which would detect whether the selected devie config changes require a hard reset, but I guess this is a good start towards that.

File: src/qt/qt_deviceconfig.cpp (hunk 1 above)
Comment (Cacodemon345) [reply src/qt/qt_deviceconfig.cpp]: I clarified it. I think the explanation I gave is good enough.

**Pull request closed without merging**